src/elm/parameters.js

```diff
@@ -13,12 +13,13 @@ class ParameterDef extends Expression {
     // If context parameters contains the name, return value.
     if (ctx && ctx.parameters[this.name] !== undefined) {
       return ctx.parameters[this.name];
+      // If the parent context contains the name, return that
+    } else if (ctx.getParentParameter(this.name) !== undefined) {
+      const parentParam = ctx.getParentParameter(this.name);
+      return parentParam.default != null ? parentParam.default.execute(ctx) : parentParam;
       // If default type exists, execute the default type
     } else if (this.default != null) {
       return this.default != null ? this.default.execute(ctx) : undefined;
```

Comment:
The check on `this.default != null` is unnecessary here since it's in a block that goes through a `this.default != null` gate.

Reply:
A good point...not sure why I would put that twice.